### ISSUE TITLE
Number string parameters get detected as date

### ISSUE THREAD

**Issue by semmel** [uskr/jsonrpc-lean#8]
When giving a number as text (e.g. `"33"`) for a request parameter the JSON parser interprets it as a date type.

E.g. (see examples/testserver.cpp)
```C++
std::string Concat(const std::string& a, const std::string& b) {
	return a + b;
}
dispatcher.AddMethod("concat", &Concat);
const char request[] = R"({"jsonrpc":"2.0","method":"concat","id":1,"params":["Hello, ","33"]})";
server.HandleRequest(concatRequest);
// -> {"jsonrpc":"2.0","id":1,"error":{"code":-32602,"message":"Invalid parameters"}}
```

`jsonrpc::util::ParseIso8601DateTime` as called from JSON reader in `Value GetValue(const rapidjson::Value&)` returns `true` which is wrong. [Live demo here](https://ideone.com/p6fBFt).

I did not expect treatment for data strings since IMO it is not part of the JSON standard. I guess interpreting strings as data and serializing dates into strings, would better done in the application layer, or I'd rather pick a whole other data format like [MessagePack](https://msgpack.org/index.html).

I would not be sad if support for `Value::Type::DATE_TIME` is simply kicked out of *jsonrpc-lean*. :expressionless:

**Comment by roqerxz** [comment]
Are you accepting pull requests for this?

**Comment by uskr** [comment]
Sorry for the late reply. This is fixed on version 1.1.0.

Agreed that it is better if we removed this part of the implementation entirely, and leave it to the application layer. In the end it is not even a type defined in the JSON RPC 2.0 specification.
